### PR TITLE
Access control via workspace ids

### PR DESCRIPTION
**Pull request by jayrbolton**

### CODE REVIEW

File: src/relation_engine_server/api.py [hunk 1]
```diff
@@ -33,9 +34,13 @@ def run_query():
...
         resp_body = arango_client.run_query(query_text=query_text, bind_vars=json_body)
         return flask.jsonify(resp_body)
-    # auth.require_auth_token(roles=[])
     if 'view' in flask.request.args:
         # Run a query from a view name
+        json_body['ws_ids'] = []
+        auth_token = auth.get_auth_header()
+        if auth_token:
+            # Handle workspace authentication
+            json_body['ws_ids'] = auth.get_workspace_ids(auth_token)
```
Comment (JamesJeffryes) [inline src/relation_engine_server/api.py]: I think we want this to run with adhoc queries as well to better represent the behavior of a defined view

Comment (jayrbolton) [reply src/relation_engine_server/api.py]: Good call on the adhoc queries, didn't consider those, will add that.

Comment (jayrbolton) [reply src/relation_engine_server/api.py]: Having read-everything admin access adds another whole layer that I'm still working on, so this PR is not really complete. Will let you know when the pr is ready again.

**Comment by jayrbolton** [comment]
Not quite ready, got pulled away from this yesterday but will finish it up this morning

**Comment by jayrbolton** [comment]
@JamesJeffryes I think this is ready now. If you can do another once-over I'd appreciate it.

**Pull request merged**